CoreAttributeEditorView/src/au/gov/asd/tac/constellation/views/attributeeditor/editors/IconEditorFactory.java

```diff
@@ -247,58 +259,85 @@ private void addNode(TreeItem<IconNode> item, IconNode node) {
         }
 
         private HBox createAddRemoveBox() {
-            //add/remove button
+            // Add/remove button.
             final HBox addRemoveBox = new HBox(BUTTON_SPACING);
             addRemoveBox.setAlignment(Pos.CENTER_RIGHT);
             final Button addFilesButton = new Button("Add File(s)...");
             final Button addDirButton = new Button("Add Directory...");
             final Button removeButton = new Button("Remove");
+
             addFilesButton.setOnAction(event -> {
-                final FileChooser addIconChooser = new FileChooser();
-                addIconChooser.setTitle("Add new Icon(s)");
-                addIconChooser.getExtensionFilters().add(new FileChooser.ExtensionFilter("PNG images", "*.*png", "*.*PNG"));
-                final List<File> selectedFiles = addIconChooser.showOpenMultipleDialog(null);
-                String lastIconName = null;
-                if (selectedFiles != null) {
+                final JFileChooser addIconChooser = new JFileChooser();
+                addIconChooser.setDialogTitle("Add New Icon(s)");
+                addIconChooser.setCurrentDirectory(SAVED_DIRECTORY != null ? SAVED_DIRECTORY : DEFAULT_DIRECTORY);
```

Comment:
Can't this be done with `FileChooserBuilder` and passing in an appropriate chooser group? It would also make it a lot cleaner. You could also test it. See https://github.com/constellation-app/constellation/blob/feature/dataaccess_packagerename/CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/components/DataAccessResultsDirChooser.java and the corresponding tests in https://github.com/constellation-app/constellation/blob/feature/dataaccess_packagerename/CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/components/DataAccessResultsDirChooserNGTest.java

Reply:
Or is the point to switch to swing?

Reply:
@formalhaunt so the point was to convert them all to one consistent style, but Swing was chosen for the modality that the JavaFX counterparts lacked. I am now converting those to the FileChooserBuilder you suggested as that retains the modality and can be tested as well shown by your examples.